src/pages/Debug/DebugDetails.tsx

```diff
@@ -0,0 +1,213 @@
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import DebugUtils from '@libs/DebugUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import ButtonWithDropdownMenu from '@src/components/ButtonWithDropdownMenu';
+import type {DropdownOption} from '@src/components/ButtonWithDropdownMenu/types';
+import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import {DETAILS_DROPDOWN_OPTIONS, DETAILS_SELECTION_LIST} from './const';
+
+type DebugDetailsProps = {
+    data: Record<string, unknown>;
+    children?: React.ReactNode;
+    /** A unique Onyx key identifying the form */
+    formId: typeof ONYXKEYS.FORMS.DEBUG_REPORT_ACTION_PAGE_FORM | typeof ONYXKEYS.FORMS.DEBUG_REPORT_PAGE_FORM;
+    reportID: string;
+    reportActionId?: string;
+    onSave: (values: FormOnyxValues<typeof ONYXKEYS.FORMS.DEBUG_REPORT_PAGE_FORM | typeof ONYXKEYS.FORMS.DEBUG_REPORT_ACTION_PAGE_FORM>) => void;
+    onDelete: () => void;
+    validate: (key: never, value: string) => void;
+};
+
+const DropdownOptionsArrays: Record<string, Array<DropdownOption<string>>> = Object.entries(DETAILS_DROPDOWN_OPTIONS).reduce((acc, [key, value]) => {
+    acc[key] = Object.values(value).map((option) => ({value: option, text: option}));
+    return acc;
+}, {} as Record<string, Array<DropdownOption<string>>>);
+
+function DebugDetails({data, children, reportID, onSave, onDelete, validate, formId, reportActionId}: DebugDetailsProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const onyxDraftDataKey = formId === ONYXKEYS.FORMS.DEBUG_REPORT_PAGE_FORM ? ONYXKEYS.FORMS.DEBUG_REPORT_PAGE_FORM_DRAFT : ONYXKEYS.FORMS.DEBUG_REPORT_ACTION_PAGE_FORM_DRAFT;
+    const [formDraftData] = useOnyx(onyxDraftDataKey);
+    const [dropdownsState, setDropdownsState] = useState<Record<string, string>>({});
+
+    const validator = useCallback(
+        (values: Record<string, unknown>): FormInputErrors<typeof ONYXKEYS.FORMS.DEBUG_REPORT_PAGE_FORM | typeof ONYXKEYS.FORMS.DEBUG_REPORT_ACTION_PAGE_FORM> => {
+            const newErrors: Record<string, string | undefined> = {};
+            Object.entries(values).forEach(([key, value]) => {
+                try {
+                    if (typeof value === 'boolean') {
+                        return;
+                    }
+
+                    validate(key as never, value as string);
+                } catch (e) {
+                    const {cause, message} = e as SyntaxError;
+                    newErrors[key] = cause || message === 'debug.missingValue' ? translate(message as TranslationPaths, cause as never) : message;
```

Comment:
I know this PR is waiting on another PR and might be waiting awhile, but while we're here can we see if we can avoid using these `as` casts? Kind of defeats the purpose of using TypeScript if we use `as` casts unless absolutely necessary.

In this case I think we could also use [type guarding](https://www.typescriptlang.org/docs/handbook/advanced-types.html) to get rid of some of the `as` casts.